### PR TITLE
Cancel a currently running workflow from the same PR when a new workflow is triggered

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
as title, always run the latest workflow in the PR.
https://stackoverflow.com/a/72408109/9574775
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by eapolinario** [comment]
This is a great feature!

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 name: Components Checks
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
```
Comment (eapolinario) [inline .github/workflows/checks.yml:4]: We need the `|| github.ref` to handle the non-PR case.

**Review by eapolinario — COMMENTED** [review]
That's a great find.

File: .github/workflows/checks.yml (hunk 1 above)
Comment (pingsutw) [reply .github/workflows/checks.yml]: updated it

**Review by eapolinario — APPROVED** [review]
Amazing!
